packages/belos/tpetra/example/BlockCG/PseudoBlockCGTpetraExFile.cpp

```diff
@@ -0,0 +1,229 @@
+    if (verbose) {
+      belosList.set( "Verbosity", Belos::Errors + Belos::Warnings +
+        Belos::TimingDetails + Belos::StatusTestDetails );
+      if (frequency > 0)
+        belosList.set( "Output Frequency", frequency );
+    }
+    else
+      belosList.set( "Verbosity", Belos::Errors + Belos::Warnings );
+    //
+    // Construct an unpreconditioned linear problem instance.
+    //
+    Belos::LinearProblem<ST,MV,OP> problem( A, X, B );
+    bool set = problem.setProblem();
+    if (set == false) {
+      if (proc_verbose)
+        std::cout << std::endl << "ERROR:  Belos::LinearProblem failed to set up correctly!" << std::endl;
+      return -1;
+    }
+    //
+    // *******************************************************************
+    // **************Start the block CG iteration*************************
+    // *******************************************************************
+    //
+    // Create an iterative solver manager.
+    RCP< Belos::SolverManager<ST,MV,OP> > newSolver
+      = rcp( new Belos::PseudoBlockCGSolMgr<ST,MV,OP>(rcp(&problem,false), rcp(&belosList,false)) );
+
+    //
+    // **********Print out information about problem*******************
+    //
+    if (proc_verbose) {
+      std::cout << std::endl << std::endl;
+      std::cout << "Dimension of matrix: " << NumGlobalElements << std::endl;
+      std::cout << "Number of right-hand sides: " << numrhs << std::endl;
+      std::cout << "Max number of CG iterations: " << maxiters << std::endl;
+      std::cout << "Relative residual tolerance: " << tol << std::endl;
+      std::cout << std::endl;
+    }
+    //
+    // Perform solve
+    //
+    Belos::ReturnType ret = newSolver->solve();
+    //
+    // Compute actual residuals.
+    //
+    bool badRes = false;
+    std::vector<ST> actual_resids( numrhs );
+    std::vector<ST> rhs_norm( numrhs );
+    MV resid(Map, numrhs);
+    OPT::Apply( *A, *X, resid );
+    MVT::MvAddMv( -1.0, resid, 1.0, *B, resid );
+    MVT::MvNorm( resid, actual_resids );
+    MVT::MvNorm( *B, rhs_norm );
+    if (proc_verbose) {
+      std::cout<< "---------- Actual Residuals (normalized) ----------"<<std::endl<<std::endl;
+      for ( int i=0; i<numrhs; i++) {
+        ST actRes = actual_resids[i]/rhs_norm[i];
+        std::cout<<"Problem "<<i<<" : \t"<< actRes <<std::endl;
+        if (actRes > tol) badRes = true;
+      }
+    }
+
+  if (ret!=Belos::Converged || badRes) {
+    success = false;
+    if (proc_verbose)
+      std::cout << std::endl << "ERROR:  Belos did not converge!" << std::endl;
+  } else {
+    success = true;
+    if (proc_verbose)
+      std::cout << std::endl << "SUCCESS:  Belos converged!" << std::endl;
+  }
+}
+TEUCHOS_STANDARD_CATCH_STATEMENTS(verbose, std::cerr, success);
+
+return success ? EXIT_SUCCESS : EXIT_FAILURE;
+}
+
+int main(int argc, char *argv[]) {
+  // run with different ST
+  run<double>(argc,argv);
+  // run<float>(argc,argv); // FAILS
+}
```

Comment:
Hello @cwschilly just a remark: the main function should return the result of the run call. Coul you please add a `return` statement to the `run` function call.

Reply:
Fixed